src/pilot/assign_role.py

```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
     return raid_controller_ids
 
 
-def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
+    all_physical_disks = drac_client.list_physical_disks()
 
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks = {}
+    for disk in all_physical_disks:
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
+                raid_cntlr_physical_disks[disk.controller].append(disk)
+            else:
+                raid_cntlr_physical_disks[disk.controller] = [disk]
 
-    # None indicates an error occurred.
-    if raid_10_logical_disk is None:
+    boss_controller = [cntrl for cntrl in raid_controllers
+                       if drac_client.is_boss_controller(cntrl)]
+
+    if boss_controller:
+        if check_cntlr_physical_disks_len(
+           raid_cntlr_physical_disks[boss_controller[0]]):
```

Comment:
Not a fan of boss_controller being a list, there can only be one boss card per system so we should just have single method _fetch_boss_card or something like that, that returns None or the boss card, not a list.

Reply:
Okay , will do this changes and push the latest changes.

Reply:
Done